addOns/client/src/main/java/org/zaproxy/addon/client/internal/ClientSideComponent.java

```diff
@@ -122,4 +121,65 @@ public boolean equals(Object obj) {
                 && Objects.equals(tagName, other.tagName)
                 && Objects.equals(text, other.text);
     }
+
+    @Override
+    public int compareTo(ClientSideComponent o) {
+        if (this.type.compareTo(o.type) > 0) {
+            return 1;
+        } else if (this.type.compareTo(o.type) < 0) {
+            return -1;
+        } else {
+            // Types are the same, Check next relevant property(ies)
+            switch (this.type) {
+                case "Cookies", "localStorage", "sessionStorage":
+                    if (this.href != null && o.href != null && this.href.compareTo(o.id) > 0) {
+                        return 1;
+                    } else if (this.href != null
+                            && o.href != null
+                            && this.href.compareTo(o.id) < 0) {
+                        return -1;
+                    } else {
+                        if (this.id.compareTo(o.id) > 0) {
+                            return 1;
+                        } else if (this.id.compareTo(o.id) < 0) {
+                            return -1;
+                        } else {
+                            return this.text.compareTo(o.text);
+                        }
+                    }
```

Comment:
I guess I should extract a method per switch case to make things easier to digest.